lib/rules/no-else-return.js

```diff
@@ -88,6 +102,34 @@ module.exports = {
                     }
 
                     const endToken = sourceCode.getLastToken(node);
+
+                    /**
+                     * If else block includes block scope local variable declaration [let or const],
+                     * then it is not safe to remove else keyword [issue 11069]
+                     */
+                    const isDeclarationInside = sourceCode.getTokensBetween(startToken, endToken).findIndex(isVariableDeclaration) > -1;
+
+                    const childScopeVars = (childScopes, store = {}, level = 0) =>
+                        childScopes.reduce((acc, block) => {
+                            if (block.childScopes.length >= 0) {
+                                acc[level] = block.variables;
+                                let index = level + 1
+                                return childScopeVars(block.childScopes, acc, index);
+                            }
+                            return acc
+                        }, store)
+
+                    const shadowedVariables = scope => {
+                        const result = childScopeVars(scope.childScopes);
+                        return scope.variables.filter(elem => result[0].findIndex(vars => vars.name === elem.name) > -1)
```

Comment:
consider using `some` instead of `findIndex` if you're only interested in a match and don't use the index